docker/Dockerfile.cross_compile_aarch64

```diff
@@ -0,0 +1,62 @@
+FROM centos:7.6.1810
+
+ARG gcc_version=4.9-2016.02
+ARG openssl_version=1_1_1d
+ARG apr_version=1.6.5
+
+ENV GCC_VERSION $gcc_version
+ENV OPENSSL_VERSION $openssl_version
+ENV APR_VERSION $apr_version
+
+# Install requirements
+RUN  set -x && \
+  yum -y install epel-release && \
+  yum -y install wget tar git make autoconf automake libtool openssl-devel ninja-build gcc-c++
+
+# Install Java and Golang
+RUN yum install -y java-1.8.0-openjdk-devel golang
+ENV JAVA_HOME="/usr/lib/jvm/java-1.8.0-openjdk/"
+
+# Install aarch64 gcc 4.9 toolchain
+RUN set -x && \
+  wget https://releases.linaro.org/components/toolchain/binaries/$GCC_VERSION/aarch64-linux-gnu/gcc-linaro-$GCC_VERSION-x86_64_aarch64-linux-gnu.tar.xz && \
+  tar xvf gcc-linaro-$GCC_VERSION-x86_64_aarch64-linux-gnu.tar.xz
+ENV PATH="/gcc-linaro-$GCC_VERSION-x86_64_aarch64-linux-gnu/bin:${PATH}"
+
+# Install CMake
+RUN yum install -y cmake3 && ln -s /usr/bin/cmake3 /usr/bin/cmake
+
+# Cross compile Apache Apr for aarch64 - static
+RUN set -x && \
+  wget https://downloads.apache.org//apr/apr-$APR_VERSION.tar.gz && \
+  tar xvf apr-$APR_VERSION.tar.gz && \
+  pushd apr-$APR_VERSION && \
+  CFLAGS='-O3 -fno-omit-frame-pointer -fPIC' ./configure --disable-shared --prefix=/opt/apr-$APR_VERSION-static --host=aarch64-linux-gnu ac_cv_file__dev_zero=yes ac_cv_func_setpgrp_void=yes apr_cv_tcp_nodelay_with_cork=yes ac_cv_sizeof_struct_iovec=8 && \
+  make || true && \
+  pushd tools && \
+  gcc -Wall -O2 -DCROSS_COMPILE gen_test_char.c -s -o gen_test_char && \
+  popd && \
+  make && make install && \
+  popd
+
+# Cross compile Apache Apr for aarch64 - share
+RUN set -x && \
+  wget https://downloads.apache.org//apr/apr-$APR_VERSION.tar.gz && \
+  tar xvf apr-$APR_VERSION.tar.gz && \
+  pushd apr-$APR_VERSION && \
+  CFLAGS='-O3 -fno-omit-frame-pointer -fPIC' ./configure --prefix=/opt/apr-$APR_VERSION-share --host=aarch64-linux-gnu ac_cv_file__dev_zero=yes ac_cv_func_setpgrp_void=yes apr_cv_tcp_nodelay_with_cork=yes ac_cv_sizeof_struct_iovec=8 && \
+  make || true && \
+  pushd tools && \
+  gcc -Wall -O2 -DCROSS_COMPILE gen_test_char.c -s -o gen_test_char && \
+  popd && \
+  make && make install && \
+  popd
+
+# Cross compile OpenSSL for aarch64 - share
+RUN set -x && \
+  wget https://github.com/openssl/openssl/archive/OpenSSL_$OPENSSL_VERSION.tar.gz && \
+  tar xvf OpenSSL_$OPENSSL_VERSION.tar.gz && \
+  pushd openssl-OpenSSL_$OPENSSL_VERSION && \
+  ./Configure linux-aarch64 --cross-compile-prefix=aarch64-linux-gnu- --prefix=/opt/openssl-$OPENSSL_VERSION-share shared && \
```

Comment:
@wangxiyuan hm... wouldn't this basically have the dynamic linked tcnative version only work if the user has openssl installed in the same path when trying to use `netty-tcnative` ?

Reply:
I don't think it has anything to do with the path used to compile the library, Linux tries multiple lib paths when loading library files or maybe I misunderstood the concern?

Reply:
@normanmaurer  Yes. If the user want to use dynamic netty-tcnative, the apr and openssl should be installed first.

While this dockerfile is just for cross compile. It's on X86 platform. So when building the code  for aarch64. The aarch64 apr/openssl should be there as well, since aarch64-gcc can't use the x86 apr/openssl when building. 

The compile error will be like:
```
/usr/include/apr-1/apr.h:21:2: error #error "This apr-devel package does not work your architecture?" 
```

So I compile the aarch64 apr/openssl here first.

Reply:
@normanmaurer 

![if-it-fits-it-ships](https://user-images.githubusercontent.com/323497/80087015-d11b4e80-855a-11ea-8a7f-ccdb6311a5e8.jpg)